File: 242-valid-anagram/242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
  if (s.length !== t.length) {
    return false;
  }
  
  const charCounts = {};
  
  for (let i = 0; i < s.length; i++) {
    charCounts[s[i]] = (charCounts[s[i]] ?? 0) + 1;
    charCounts[t[i]] = (charCounts[t[i]] ?? 0) - 1;
  }
  
  return Object.values(charCounts).every((count) => count === 0);
};

File: 242-valid-anagram/242-valid-anagram.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }
    
    runeCounts := make(map[byte]int)
    
    for i := 0; i < len(s); i++ {
        runeCounts[s[i]] += 1
        runeCounts[t[i]] -= 1
    }
    
    for _, count := range runeCounts {
        if count != 0 {
            return false
        }
    }
    return true
}